packages/css-data/src/property-parsers/box-shadow-property-extractor.ts

```diff
@@ -0,0 +1,51 @@
+import type {
+  KeywordValue,
+  RgbValue,
+  TupleValue,
+  TupleValueItem,
+  UnitValue,
+} from "@webstudio-is/css-engine";
+
+export const extractBoxShadowProperties = (
+  shadow: TupleValue
+): {
+  inset: KeywordValue | null;
+  color: RgbValue | KeywordValue | null;
+  offsetX: UnitValue | null;
+  offsetY: UnitValue | null;
+  blur: UnitValue | null;
+  spread: UnitValue | null;
+} => {
+  let properties = [...shadow.value];
+  let inset: KeywordValue | null = null;
+  let color: KeywordValue | RgbValue | null = null;
+
+  properties = properties.filter(
+    (prop: TupleValueItem): prop is KeywordValue => {
+      if (prop.type === "keyword" && prop.value === "inset") {
+        inset = prop;
+        return false;
+      }
+
+      if (prop.type === "keyword" || prop.type === "rgb") {
+        color = prop;
+        return false;
+      }
+
+      return true;
+    }
+  );
+
+  const [offsetX, offsetY, blur, spread] = properties.filter(
+    (item): item is UnitValue => item.type === "unit"
+  );
```

Comment:
Based on spec 
```
<shadow> = 
  <color>?                                   &&
  [ <length>{2} <length [0,∞]>? <length>? ]  &&
  inset?                                     
```

So color can't be between length's same as inset

This `2px 2px rgba(255, 0, 0, 1) 2px 1px ` should not be parsed as 
color = rgba(255, 0, 0, 1)

Reply:
Yes, but in this case. The `proeprty-parser/box-shadow` take care of all such errors and generates this tuple. So, it's safe the tuple that is generated here is always a valid one. Or else the step where users enter such invalid `2px 2px rgba(255, 0, 0, 1) 2px 1px ` entries will get errors.

This step was just to assign them independently 👍

Reply:
The idea of functions is to not assume how the system around are working ;-)

Reply:
In real-word use-case. It can be something like

```js
const { blur, spread } = extractBoxShadowProeprties(parseBoxShadow("2px 2px rgba(255, 0, 0, 1) 2px 1px)
```

Something like this.

Reply:
I get it, we need to make them pure without assumptions. But, just curios. If i can find the spec with index matching and throw errors 👍

Reply:
specs are unreadable disaster ;-)